version: '3'

networks:
  transcendence:
    name: transcendence

volumes:
  postgres_data:
    driver: local

services:
  django:
    build:
      context: ./${BACK_END_FOLDER}/
      dockerfile: Dockerfile-django
    env_file: .env
    ports:
      - "8000:8000"
    depends_on:
      - postgres
    networks:
      - transcendence
    restart: on-failure

  daphne:
    build:
      context: ./${BACK_END_FOLDER}/
      dockerfile: Dockerfile-daphne
    env_file: .env
    ports:
      - "8001:8001"
    depends_on:
      - django
    networks:
      - transcendence
    restart: on-failure

  nginx:
    build:
      context: ./${NGINX_FOLDER}/
      dockerfile: Dockerfile
    env_file: .env
    ports:
      - "80:80"
      - "443:443"
    depends_on:
      - daphne
    networks:
      - transcendence
    restart: on-failure

  postgres:
    image: postgres:latest
    env_file: .env
    environment:
      POSTGRES_DB: ${POSTGRES_DB}
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    ports:
      - "5432:5432"
    networks:
      - transcendence
    restart: on-failure

  adminer:
    image: adminer
    restart: always
    ports:
      - "8080:8080"
    networks:
      - transcendence